Cannot save game anymore since v0.19.1

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong
Hi,
I cannot save my game any more since version v0.19.1. Deleting the configuration folder (including ja2.json) did not bring any benefits. However, loading (old) saved games still works. 

An error message occurs while saving:
2022-05-27T08:08:16 [ERROR] game\SaveLoadGame.cc: Error saving game: FileMan::moveFile('C:\Users\user\AppData\Local\Temp\ja2-stracciatella-QdUo8WdF\save\QuickSave.sav', 'X:\home\Documents\JA2\SavedGames\QuickSave.sav') failed: Fs_rename "C:\\Users\\user\\AppData\\Local\\Temp\\ja2-stracciatella-QdUo8WdF\\save\\QuickSave.sav" "X:\\home\\Documents\\JA2\\SavedGames\\QuickSave.sav": Das System kann die Datei nicht auf ein anderes Laufwerk verschieben. (os error 17)

The operating system is Windows 10 21H2.

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ---------------------- | ------------------- | ------ | -------- |
|  v0.19.1-git+6ad1b6f | JA2 Gold German  |    no    |    yes     |



[//]: # (Please also include:)
[//]: # (- how to reproduce the bug and, if possible, a save game)
[//]: # (- the operating system)
[//]: # (- in case you know, for enhancements, if this feature was already added by some other source mod like 1.13)

[//]: # (When pasting backtraces or any other logs with hashed enumerations, please format it as a code block.)
[//]: # (You can do that by selecting it and clicking the "<>" icon or manually putting three backticks [```] at each end.)

--- a/rust/stracciatella/src/fs.rs
+++ b/rust/stracciatella/src/fs.rs
@@ -63,6 +63,12 @@ pub fn rename<P: AsRef<Path>>(from: P, to: P) -> Result<(), io::Error> {
         Err(e) if e.raw_os_error() == Some(libc::EXDEV) => {
             std::fs::copy(&from, &to).and_then(|_| remove_file(&from))
         }
+        #[cfg(target_family = "windows")]
+        Err(e)
+            if e.raw_os_error() == Some(winapi::shared::winerror::ERROR_NOT_SAME_DEVICE as i32) =>
+        {
+            std::fs::copy(&from, &to).and_then(|_| remove_file(&from))
+        }
         Err(e) => Err(e),
     }
 }

